### PR TITLE
FreeBSD cleanup

### PR DESCRIPTION
**Pull request by ryao**
### Motivation and Context
FreeBSD's coverity scans complained about a few things that are not real issues, but are cleanup opportunities.

### Description
We do cleanup.

### How Has This Been Tested?
The buildbot can test it.

### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Performance enhancement (non-breaking change which improves efficiency)
- [x] Code cleanup (non-breaking change which makes code smaller or more readable)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Library ABI change (libzfs, libzfs\_core, libnvpair, libuutil and libzfsbootenv)
- [ ] Documentation (a change to man pages or other documentation)

### Checklist:

- [x] My code follows the OpenZFS [code style requirements](https://github.com/openzfs/zfs/blob/master/.github/CONTRIBUTING.md#coding-conventions).
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**contributing** document](https://github.com/openzfs/zfs/blob/master/.github/CONTRIBUTING.md).
- [ ] I have added [tests](https://github.com/openzfs/zfs/tree/master/tests) to cover my changes.
- [ ] I have run the ZFS Test Suite with this change applied.
- [x] All commit messages are properly formatted and contain [`Signed-off-by`](https://github.com/openzfs/zfs/blob/master/.github/CONTRIBUTING.md#signed-off-by).

### CODE REVIEW

**Review by szubersk — APPROVED** [review]

**Review by freqlabs — APPROVED** [review]

File: lib/libzfs/os/freebsd/libzfs_zmount.c [hunk 1]
```diff
@@ -116,7 +116,7 @@ do_mount(zfs_handle_t *zhp, const char *mntpt, const char *opts, int flags)
 {
 
 	return (do_mount_(zfs_get_name(zhp), mntpt, flags, NULL, 0,
-	    opts, sizeof (mntpt)));
+	    opts, MNT_LINE_MAX));
```
Comment (freqlabs) [inline lib/libzfs/os/freebsd/libzfs_zmount.c:119]: Really seems like `do_mount_()` doesn't even need to be a separate function.

Comment (ryao) [reply lib/libzfs/os/freebsd/libzfs_zmount.c:119]: Good point. I will combine them and repush when I get a chance.

**Review by amotin — APPROVED** [review]

File: lib/libzfs/os/freebsd/libzfs_zmount.c (hunk 1 above)
Comment (ryao) [reply lib/libzfs/os/freebsd/libzfs_zmount.c]: I have combined them and repushed.

**Review by amotin — APPROVED** [review]

**Review by szubersk — APPROVED** [review]

**Review by behlendorf — CHANGES_REQUESTED** [review]
These changes appear to break the build on FreeBSD.

**Comment by ryao** [comment]
@behlendorf I have pushed a revised commit that fixes those typos. It should build now. I am relying on the buildbot to tell me whether it builds, but was too busy to see the results before you did.

**Review by behlendorf — APPROVED** [review]

**Pull request closed without merging**